File: windows/net/samples.vb/MultiDataDisc/WriteSpeedsForm.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace MultiDataDisc
	Partial Public Class WriteSpeedsForm
		Inherits Form

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub buttonOK_Click(ByVal sender As Object, ByVal e As EventArgs) Handles buttonOK.Click
			_selectedWriteSpeed = TryCast(cbWriteSpeed.SelectedItem, SpeedInfo)
			DialogResult = System.Windows.Forms.DialogResult.OK
		End Sub

		Private Sub buttonCancel_Click(ByVal sender As Object, ByVal e As EventArgs) Handles buttonCancel.Click
			DialogResult = System.Windows.Forms.DialogResult.Cancel
		End Sub

		Public Property WriterTitle() As String
			Get
				Return lblWriter.Text
			End Get

			Set(ByVal value As String)
				lblWriter.Text = value
			End Set
		End Property

		Private _writeSpeeds As List(Of SpeedInfo)
		Public Property WriteSpeeds() As List(Of SpeedInfo)
			Get
				Return _writeSpeeds
			End Get
			Set(ByVal value As List(Of SpeedInfo))
				_writeSpeeds = value
				UpdateWriteSpeeds()
			End Set
		End Property

		Private _selectedWriteSpeed As SpeedInfo
		Public Property SelectedWriteSpeed() As SpeedInfo
			Get
				Return _selectedWriteSpeed
			End Get
			Set(ByVal value As SpeedInfo)
				_selectedWriteSpeed = value
				UpdateWriteSpeeds()
			End Set
		End Property

		Private Sub UpdateWriteSpeeds()
			cbWriteSpeed.Items.Clear()

			If _writeSpeeds IsNot Nothing Then
				For i As Integer = 0 To _writeSpeeds.Count - 1
					cbWriteSpeed.Items.Add(_writeSpeeds(i))
				Next i
			End If

			If _selectedWriteSpeed IsNot Nothing Then
				If cbWriteSpeed.Items.Count > 0 Then
					cbWriteSpeed.SelectedIndex = cbWriteSpeed.FindString(_selectedWriteSpeed.ToString())

					If -1 = cbWriteSpeed.SelectedIndex Then
						cbWriteSpeed.SelectedIndex = 0
					End If
				End If
			End If
		End Sub
	End Class
End Namespace

File: windows/net/samples/MultiDataDisc/WriteSpeedsForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace MultiDataDisc
{
    public partial class WriteSpeedsForm : Form
    {
        public WriteSpeedsForm()
        {
            InitializeComponent();
        }

        private void buttonOK_Click(object sender, EventArgs e)
        {
            _selectedWriteSpeed = cbWriteSpeed.SelectedItem as SpeedInfo;
            DialogResult = DialogResult.OK;
        }

        private void buttonCancel_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
        }

        public string WriterTitle
        {
            get
            {
                return lblWriter.Text;
            }

            set
            {
                lblWriter.Text = value;
            }
        }

        List<SpeedInfo> _writeSpeeds;
        public List<SpeedInfo> WriteSpeeds
        {
            get { return _writeSpeeds; }
            set { _writeSpeeds = value; UpdateWriteSpeeds(); } 
        }

        SpeedInfo _selectedWriteSpeed;
        public SpeedInfo SelectedWriteSpeed 
        {
            get { return _selectedWriteSpeed; }
            set { _selectedWriteSpeed = value; UpdateWriteSpeeds();  } 
        }

        void UpdateWriteSpeeds()
        {
            cbWriteSpeed.Items.Clear();

            if (_writeSpeeds != null)
            {
                for (int i = 0; i < _writeSpeeds.Count; i++)
                    cbWriteSpeed.Items.Add(_writeSpeeds[i]);
            }

            if (_selectedWriteSpeed != null)
            {
                if (cbWriteSpeed.Items.Count > 0)
                {
                    cbWriteSpeed.SelectedIndex = cbWriteSpeed.FindString(_selectedWriteSpeed.ToString());

                    if (-1 == cbWriteSpeed.SelectedIndex)
                        cbWriteSpeed.SelectedIndex = 0;
                }
            }
        }
    }
}
